test/src/main/java/org/apache/accumulo/test/ample/usage/TabletFileUpdateIT.java

```diff
@@ -0,0 +1,268 @@
+    String metadataTable = tableNames[0];
+
+    var tabletTime = TabletTime.getInstance(new MetadataTime(1, TimeType.LOGICAL));
+    long flushNonce = 42L;
+
+    try (var client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(metadataTable);
+      TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      Ample testAmple = TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      assertNull(testAmple.readTablet(extent));
+
+      // create tablet in the fake metadata table
+      testAmple.mutateTablet(extent).putDirName("dir1").putTime(tabletTime.getMetadataTime())
+          .putLocation(Location.current(tserverInstance)).putPrevEndRow(null).mutate();
+
+      // get the tablets metadata
+      var lastMetadata = testAmple.readTablet(extent);
+
+      testAmple.mutateTablet(extent).deleteLocation(Location.current(tserverInstance)).mutate();
+
+      // try locations that differ in type, port, and session
+      for (var location : List.of(Location.future(tserverInstance),
+          Location.current(new TServerInstance("localhost:9998", 0xabcdef123L)),
+          Location.current(new TServerInstance("localhost:9997", 0xabcdef124L)))) {
+        // set a location on the tablet that will not match
+        testAmple.mutateTablet(extent).putLocation(location).mutate();
+        // should fail to add file to tablet because tablet location is not as expected
+        assertThrows(IllegalStateException.class,
+            () -> updateTabletDataFile(testAmple, tabletTime.getTime(), newFile, dfv1, Set.of(), 7L,
+                MinorCompactionReason.SYSTEM, tserverInstance, extent, lastMetadata, tabletTime,
+                flushNonce));
+        // verify that files was not added
+        var updatedMetadata = testAmple.readTablet(extent);
+        assertEquals(Set.of(), updatedMetadata.getFiles());
+        testAmple.mutateTablet(extent).deleteLocation(location).mutate();
+      }
+
+      client.tableOperations().delete(metadataTable);
+    }
+  }
+
+  /**
+   * This test exercises conditional mutations returning unknown and this causing reading of the
+   * flush nonce.
+   */
+  @Test
+  public void testFlushNonce() throws Exception {
+    String[] tableNames = getUniqueNames(1);
+    String metadataTable = tableNames[0];
+
+    var tabletTime = TabletTime.getInstance(new MetadataTime(1, TimeType.LOGICAL));
+    long flushNonce = 42L;
+
+    try (var client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(metadataTable);
+      TestAmple.create(getCluster().getServerContext(),
+          Map.of(Ample.DataLevel.USER, metadataTable));
+
+      // create test ample that will return unknown status for conditional mutations.
+      Ample testAmple = TestAmple.create(getCluster().getServerContext(),
```

Comment:
It's nice that we can easily test the rejection handlers with different simulated states now that we have `TestAmple`, hopefully we can go back and add some more tests like this for other rejection handlers over time.